server/capabilities/gke_version_endpoint.go

```diff
@@ -0,0 +1,173 @@
+package capabilities
+
+import (
+	"net/http"
+	"os"
+	"strings"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"context"
+	"google.golang.org/api/container/v1"
+	"sync"
+	"io/ioutil"
+	"io"
+	"encoding/json"
+	"fmt"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	defaultCredentialEnv = "GOOGLE_APPLICATION_CREDENTIALS"
+)
+
+var mutex = &sync.Mutex{}
+
+func NewGKECapabilitiesHandler() *gkeVersionHandler {
+	return &gkeVersionHandler{}
+}
+
+type gkeVersionHandler struct {
+}
+
+type errorResponse struct {
+	Error string `json:"error"`
+}
+
+type requestBody struct {
+	Key       string `json:"key"`
```

Comment:
Change key to `credential` to match the field in `googleKubernetesEngineConfig`?  Otherwise basic functionality LGTM.